src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.revature.services;
+
+import java.util.ArrayList;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.revature.domain.QuestionPool;
+import com.revature.repositories.QuestionRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
+public class QuestionLogicImpl implements QuestionLogic{
+	
+	@Autowired
+	private QuestionRepository dao;
+
+	@Override
+	public Page<QuestionPool> getAllQuestions(Pageable pageable, String subject) {
+		
+		if(subject.equalsIgnoreCase("all")){
+			return dao.findAll(pageable);
+		} else {
+			return dao.findAllBySubjectSubjectIgnoreCase(pageable, subject);
+		}
+	}
+	
+	@Override
+	public Page<QuestionPool> searchQuestions(Pageable pageable, String text, String subject) {
+		
+		if(subject.equalsIgnoreCase("all")){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the "all" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1132)